ldms/src/ldmsd/ldmsd_cfgobj.c

```diff
@@ -301,7 +301,7 @@ void ldmsd_cfgobj_put(ldmsd_cfgobj_t obj, const char *ref_name)
 
 void ldmsd_cfgobj_del(ldmsd_cfgobj_t obj)
 {
-	pthread_mutex_lock(cfgobj_locks[obj->type]);
+	ldmsd_cfgobj_rm(obj);
```

Comment:
@morrone @tom95858 

I agree that having `ldmsd_cfgobj_del()` call `ldmsd_cfgobj_rm()` to remove the object from the tree is cleaner.

I think we should also move the cleanup logic for `obj->avl_str` and `obj->kvl_str` (lines 305-312) into `ldmsd_cfgobj___del()` as part of this PR. This change causes the cleanup to occur outside any lock protection. Moving it to `ldmsd_cfgobj___del()` ensures these fields are freed when the reference count reaches zero, which is the proper protection mechanism. It also keeps the related cleanup logic changes together in one place. What do you think?